nuxeo-features/nuxeo-platform-userworkspace/nuxeo-platform-userworkspace-core/src/test/java/org/nuxeo/ecm/platform/userworkspace/core/tests/TestUserWorkspace.java

```diff
@@ -337,7 +336,28 @@ public void testCannotRetrieveUserWorkspaceWithoutDomains() {
         session.removeDocuments(refs.toArray(new DocumentRef[refs.size()]));
         session.save();
         try (CloseableCoreSession userSession = coreFeature.openCoreSession("toto")) {
-            DocumentModel uw = uwm.getCurrentUserPersonalWorkspace(userSession, null);
+            DocumentModel uw = uwm.getCurrentUserPersonalWorkspace(userSession);
+            assertNull(uw);
+        }
+    }
+
+    /**
+     * @since 10.3
+     */
+    @Test
+    public void testCanRetrieveUserWorkspaceWithTrashedDomain() {
+        try (CloseableCoreSession userSession = coreFeature.openCoreSession("toto")) {
+            DocumentModel uw = uwm.getCurrentUserPersonalWorkspace(userSession);
+            assertNotNull(uw);
+        }
+        List<DocumentModel> docs = session.getChildren(session.getRootDocument().getRef())
+                                        .stream()
+                                        .collect(Collectors.toList());
+        TrashService trashService = Framework.getService(TrashService.class);
+        trashService.trashDocuments(docs);
```

Comment:
Coule you also add code in this test method to untrash and check you find the userworkspace again?